docs/maintainers/releasing.md

```diff
@@ -1,77 +1,74 @@
-The summary (first line) of the annotated tag message should be the full release title,
-e.g. 'v6.0.0-alpha.7 release of PowerShell'.
-
-While creating a release, it is advised to make a new branch such that
-necessary documentation updates and hot fixes can be made,
-without having to include all changes made to master.
-This release branch can be reviewed by the normal PR process.
-
-When the annotated tag is finalized, push it with `git push --tags`.
-GitHub will see the tag and present it as an option when creating a new [release][].
-Start the release, use the annotated tag's summary as the title,
-and save the release as a draft while you upload the binary packages.
-
-Just as important as creating the release is updating the links on our readme,
-and the package names in the installation instructions.
-The AppVeyor build number should also be incremented.
-
-When creating the packages, please ensure that the file path does not contain user names.
+# PowerShell Core Releasing Process
+
+## Release Steps
+
+1. Create a branch named `release` in `PowerShell/PowerShell` repository. All release related changes should happen in this branch.
+1. Prepare packages
+   - [Build release packages](#building-packages)
+   - Sign the MSI packages and DEB/RPM packages.
+   - Install and verify the packages. (_could be automated [Issue #3942][]_)
+1. Update documentation, scripts and Dockerfiles
+   - Summarize the change log for the release. It should be reviewed by PM(s) to make it more user-friendly.
+   - Update [CHANGELOG.md](../../CHANGELOG.md) with the finalized change log draft.
+   - Update other documents and scripts to use the new package names and links.
+1. Verify the release Dockerfiles. (_could be automated [Issue #3942][]_)
+1. Publish Linux packages to Microsoft YUM/APT repositories.
+1. [Create NuGet packages](#nuget-packages) and publish them to [powershell-core feed][ps-core-feed].
+1. [Create the release tag](#release-tag) and push the tag to `PowerShell/PowerShell` repository.
+1. Merge the `release` branch to `master` and delete the `release` branch.
+1. Publish the release in Github.
+1. Trigger the release docker builds for Linux and Windows container images.
+   - Linux: push a branch named `docker` to `powershell/powershell` repository to trigger the build at [powershell docker hub](https://hub.docker.com/r/microsoft/powershell/builds/).
+     Delete the `docker` branch once the builds succeed.
+   - Windows: queue a new build in `PowerShell Windows Docker Build` on VSTS.
+1. Verify the generated docker container images. (_could be automated [Issue #3942][]_)
+1. [Update the homebrew formula](#homebrew) for the OSX package.
+   This task usually will be taken care of by the community,
+   so we can wait for one day or two and see if the homebrew formula has already been updated,
+   and only do the update if it hasn't.
+
+> Note: Step 2, 3 and 4 can be done in parallel. Step 5 and 6 can be done in parallel.
```

Comment:
I think it's good enough if it stands out in the rendered Markdown file, because once we sort out the steps and the order, I don't expect this section to be changed often. But if there is a better way, I'm surely open to it.